hub/core/chunk_engine/write.py

```diff
@@ -0,0 +1,167 @@
+import numpy as np
+import pickle
+from uuid import uuid1
+
+from hub.core.chunk_engine import generate_chunks
+
+from hub.core.typing import Provider
+from typing import Any, Callable, List, Tuple
+
+from .util import (
+    array_to_bytes,
+    normalize_and_batchify_shape,
+    get_meta_key,
+    get_index_map_key,
+    get_chunk_key,
+)
+
+
+def write_array(
+    array: np.ndarray,
+    key: str,
+    chunk_size: int,
+    storage: Provider,
+    batched: bool = False,
+    tobytes: Callable[[np.ndarray], bytes] = array_to_bytes,
+):
+    """Chunk & write an array to storage.
+
+    For more on chunking, see the `generate_chunks` method.
+
+    Args:
+        array (np.ndarray): Array to be chunked/written. Batch axis (`array.shape[0]`) is optional, if `array` does have a
+            batch dimension, you should pass the argument `batched=True`.
+        key (str): Key for where the chunks, index_map, & meta will be located in `storage` relative to it's root.
+        chunk_size (int): Desired length of each chunk.
+        storage (Provider): Provider for storing the chunks, index_map, & meta.
+        batched (bool): If True, the provied `array`'s first axis (`shape[0]`) will be considered it's batch axis.
```

Comment:
"batch axis" needs to be changed or defined. It's not a standard term as far as I can tell. I like "primary axis", but whatever we go with it should be the same everywhere

Reply:
the definition is in the "batched" arg description:

> the provied `array`'s first axis (`shape[0]`) will be considered it's batch axis

Reply:
ill rename it to primary axis

Reply:
because batch usually means a fixed-sized subset of the data

Reply:
in write_array it should be called `batch_axis` because it is quite literally just a batch.

however, write_bytes it should be called `primary_axis` because it is no longer a batch of samples, but it is just a flattened sample